src/latent_gp.jl

```diff
@@ -0,0 +1,29 @@
+"""
+    LatentGP(f<:GP, x<:AbstractVector, Φ)
+
+ - `fx` is a `FiniteGP`.
+ - `Φ` is the log likelihood function which maps input of type `Tx` to `Real`.
+    
+"""
+struct LatentGP{T<:AbstractGPs.FiniteGP}
+    fx::T
+    Φ
+end
+
+function Distributions.rand(rng::AbstractRNG, lgp::LatentGP)
+    f = rand(rng, lgp.fx)
+    y = rand(rng, lgp.Φ(f))
+    return (f=f, y=y)
+end
```

Comment:
I think this would be nice to create `rand` functions also to just get `f` or just get `y`.
Like
```
randf (rng::AbstractRNG, lgp::LatentGP) = rand(rng, lgp.fx)
randy(rng::AbstractRNG, lgp::LatentGP) = rand(rng, lgp).y
```

Reply:
I disagree. What's wrong with
```julia
rand(rng, lgp).f
rand(rng, lgp).y
```
?

Reply:
If you are only interested in `f` you allocate `y` for nothing. But I agree that `randy` is not needed

Reply:
Sure. If you're worried about performance you can just call
```julia
rand(rng, lgp.fx)
```

Reply:
This is not resolved, I really can't see the advantage of having `randf` and `randy`.

Reply:
I agree we can forget `randy`, but I would still insist that `randf` can be a useful wrapper.

Reply:
But why?

Reply:
After a second look, I too feel it is better not have `randf` or `randy`. All the components required are exposed and (if required) we could mention this usage in the docstring.